ffmuc-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -156,7 +213,64 @@ if [ "$ret" -eq 1 ]; then
 else
 	PROTO=https
 fi
-force_wan_connection wget -q  -O- --post-data='{"domain": "'"$SEGMENT"'","public_key": "'"$PUBLICKEY"'"}' "$PROTO://$(uci get wireguard.mesh_vpn.broker)"
+
+# Remove API path suffix if still present in config
+WGKEX_BROKER_BASE_PATH="$(get_site_string mesh_vpn.wireguard.broker | sed 's|/api/v1/wg/key/exchange||')"
+
+if is_loadbalancing_enabled; then
+	# Use /api/v2, get gateway peer details from broker response
+	WGKEX_BROKER="$PROTO://$WGKEX_BROKER_BASE_PATH/api/v2/wg/key/exchange"
+	logger -p info -t checkuplink "Loadbalancing enabled. Contacting wgkex broker $WGKEX_BROKER"
+	if ! WGKEX_DATA=$(force_wan_connection wget -q -O- --post-data='{"domain": "'"$SEGMENT"'","public_key": "'"$PUBLICKEY"'"}' "$WGKEX_BROKER"); then
+		logger -p err -t checkuplink "Contacting wgkex broker failed, response: $WGKEX_DATA"
+		exit 1
+	fi
+
+	logger -p info -t checkuplink "Got data from wgkex broker: $WGKEX_DATA"
+
+	# Parse the returned JSON in a Lua script, returning the endpoint address, port, pubkey and first allowed IP, separated by newlines
+	if ! data=$(lua /lib/gluon/gluon-mesh-wireguard-vxlan/parse-wgkex-response.lua "$WGKEX_DATA"); then
+		logger -p err -t checkuplink "Parsing wgkex broker data failed"
+		exit 1
```

Comment:
I would eventually handle the case of "Parsing wgkex broker data failed" and fall back to v1 instead of not connecting at all?

Reply:
Sounds like a good idea :).

Reply:
@grische @dasskelett some thing to consider?

Reply:
i tried to move the call for v1 and v2 in own functions.
So v2 can call v1 when it failes:

https://github.com/DasSkelett/community-packages/pull/1/files
Here is a logread from the checkuplink:
https://gist.github.com/T0biii/1c14b56fc8488cae95181cfa359f53e0

Reply:
@maurerle thanks a lot, that's a very good idea. I added the patches + some suggestions of mine on top of the current PR.